zcash_client_backend/src/data_api/testing/pool.rs

```diff
@@ -3226,6 +3233,109 @@ pub fn change_note_spends_succeed<T: ShieldedPoolTester>(
     );
 }
 
+pub fn cross_account_transfer<T: ShieldedPoolTester, DSF>(ds_factory: DSF, cache: impl TestCache)
+where
+    DSF: DataStoreFactory,
+    <DSF as DataStoreFactory>::DataStore: Reset,
+{
+    let mut st = TestBuilder::new()
+        .with_data_store_factory(ds_factory)
+        .with_block_cache(cache)
+        .build();
+
+    // Add two accounts to the wallet.
+    let seed = Secret::new([0u8; 32].to_vec());
+    let birthday = AccountBirthday::from_sapling_activation(st.network(), BlockHash([0; 32]));
+    let (account1, usk) = st
+        .wallet_mut()
+        .create_account("account1", &seed, &birthday, None)
+        .unwrap();
+    let dfvk = T::sk_to_fvk(T::usk_to_sk(&usk));
+
+    let (account2, usk2) = st
+        .wallet_mut()
+        .create_account("account2", &seed, &birthday, None)
+        .unwrap();
+    let dfvk2 = T::sk_to_fvk(T::usk_to_sk(&usk2));
+
+    // Add funds to the account 0 in a single note
+    let value = Zatoshis::from_u64(100000).unwrap();
+    let (h, b0_result, _) = st.generate_next_block(&dfvk, AddressType::DefaultExternal, value);
+    st.scan_cached_blocks(h, 1);
+    let txid0 = *b0_result
+        .txids()
+        .first()
+        .expect("A transaction was created.");
+
+    // Spendable balance matches total balance
+    assert_eq!(st.get_total_balance(account1), value);
+    assert_eq!(
+        st.get_spendable_balance(account1, ConfirmationsPolicy::MIN),
+        value
+    );
+    assert_eq!(st.get_total_balance(account2), Zatoshis::ZERO);
+
+    let amount_sent = Zatoshis::from_u64(20000).unwrap();
+    let addr2 = T::fvk_default_address(&dfvk2);
+    let req = TransactionRequest::new(vec![
+        // payment to an account 2
+        Payment::without_memo(addr2.to_zcash_address(st.network()), amount_sent),
+    ])
+    .unwrap();
+
+    let change_strategy = fees::standard::SingleOutputChangeStrategy::new(
+        StandardFeeRule::Zip317,
+        None,
+        T::SHIELDED_PROTOCOL,
+        DustOutputPolicy::default(),
+    );
+    let input_selector = GreedyInputSelector::new();
+
+    let txid1 = st
+        .spend(
+            &input_selector,
+            &change_strategy,
+            &usk,
+            req,
+            OvkPolicy::Sender,
+            ConfirmationsPolicy::MIN,
+        )
+        .unwrap()[0];
+
+    let amount_left = (value - (amount_sent + MINIMUM_FEE).unwrap()).unwrap();
+    assert_eq!(st.get_total_balance(account1), amount_left);
+
+    let (h, _) = st.generate_next_block_including(txid1);
+    st.scan_cached_blocks(h, 1);
+
+    assert_eq!(st.get_total_balance(account2), amount_sent);
+    assert_eq!(st.get_total_balance(account1), amount_left);
+
+    // txid0 should exist; we haven't enhanced it so we'll have the mined height, but not the raw
+    // transaction data.
+    assert_matches!(st.wallet_mut().get_tx_height(txid0), Ok(Some(_)));
+
+    // delete account 1
+    assert_matches!(st.wallet_mut().delete_account(account1), Ok(_));
```

Comment:
Re: my bug comment, this tests deleting the source account in a cross-account transfer, but we need a test of deleting the destination account as well.